echemdb/data/cv/entry.py

```diff
@@ -100,40 +100,123 @@ def __getitem__(self, name):
         """
         return Descriptor(self.package.descriptor)[name]
 
-    def df(self, yunit=None):
+    def xy_units(self, xunit=None, yunit=None):
+        r"""Returns a set of x and y units as astropy units, where the resulting xunit has the magnitude of 
+        volt (V) and the resulting yunit has the magnitude of a current (I) or a current density (j).
+
+        Whether or not the yunit is in I or j, is determined from the column names of the data frame.
+
+        Examples:
+
+        When the parameters for `xunit` and `yunit` parameters are `None`, SI units are returned.::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.xy_units()
+            (Unit("V"), Unit("A / m2"))
+
+        When the parameter is set to 'original', the original values of the publisged figure are returned.::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.xy_units(xunit='original', yunit='original')
+            (Unit("V"), Unit("mA / cm2"))
+
+        The resulting units can also be set separately.::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.xy_units(xunit='original', yunit='uA / cm2')
+            (Unit("V"), Unit("uA / cm2"))
+
+        """
+        import pandas as pd
+        from astropy import units as u
+
+        column_names = pd.read_csv(self.package.resources[0].raw_iter(stream=False)).columns
+        
+        if yunit is None:
+            if 'j' in column_names:
+                yunit = u.A / u.m**2
+            if 'I' in column_names:
+                yunit = u.A
```

Comment:
This if-j-if-I is really annoying.

Could we do the following: We have `x_unit` and `y_unit` which return the units of the axes, as you planned here. Then we have `x` and `y` which return the name of the axes, i.e., `U` and `I`/`j`.

```suggestion
            if self.y == 'j':
                yunit = u.A / u.m**2
            elif self.y == 'I':
                yunit = u.A
            else:
                raise NotImplementedError("Unexpected naming of y axis.")
```

Also, we should always catch unexpected values as early as possible, as this makes debugging much easier when something goes wrong.